server/expression.js

```diff
@@ -357,12 +373,15 @@ expression.readMatrix_ = function(fileName, inputGenes,
     });
   }
   geneNames.forEach(function(geneName) {
-    var geneIndex = allGeneNames[geneName];
-    if (geneName in allGeneNames) {
+    var geneIndex = allGeneNames[geneName.toLowerCase()].index;
+    if (geneName.toLowerCase() in allGeneNames) {
       var parts = lines[geneIndex].split(/[\t\s]+/);
       var tmpLine = [];
       conditions.forEach(function(conditionIndex) {
         var value = parseFloat(parts[conditionIndex]);
+        if (!value) {
```

Comment:
I replaced the outdated binary file with the up-to-date text file, finding out there are no more neither empty nor undefined values. But I think we had better keep the value validation part to ensure valueMin/valueMax are not undefined all the time. Because if there occurs any type or other unaware mistakes in the dataset, resulting in any value was undefined, valueMin/valueMax would become undefined and the whole line chart would break down. I think it is necessary to allow some minor mistakes in the dataset and not break down the whole chart because of them.